dll/win32/fmifs/format.c

```diff
@@ -65,20 +66,37 @@ FormatEx(
         return;
     }
 
-#if 1
-    DPRINT1("Warning: use GetVolumeNameForVolumeMountPointW() instead!\n");
-    swprintf(VolumeName, L"\\??\\%c:", towupper(DriveRoot[0]));
-    RtlCreateUnicodeString(&usDriveRoot, VolumeName);
-    /* Code disabled as long as our storage stack doesn't understand IOCTL_MOUNTDEV_QUERY_DEVICE_NAME */
-#else
-    if (!GetVolumeNameForVolumeMountPointW(DriveRoot, VolumeName, RTL_NUMBER_OF(VolumeName)) ||
-        !RtlDosPathNameToNtPathName_U(VolumeName, &usDriveRoot, NULL, &CurDir))
+    if (!NT_SUCCESS(RtlStringCchCopyW(DriveName, ARRAYSIZE(DriveName), DriveRoot)))
     {
-        /* Report an error */
         Callback(DONE, 0, &Success);
         return;
     }
-#endif
+
+    if (DriveName[wcslen(DriveName) - 1] != L'\\')
+    {
+        /* Append the trailing backslash for GetVolumeNameForVolumeMountPointW */
+        if (!NT_SUCCESS(RtlStringCchCatW(DriveName, ARRAYSIZE(DriveName), L"\\")))
+        {
+            Callback(DONE, 0, &Success);
+            return;
+        }
+    }
+
+    if (!GetVolumeNameForVolumeMountPointW(DriveName, VolumeName, ARRAYSIZE(VolumeName)))
```

Comment:
Why ARRAYSIZE and not _countof here and in other places?

Reply:
Requires including an additional that defines it, is changing that really necessary for anything?

Reply:
countof requires stdlib.h indeed. If it's not included, just keep continuing using arraysize instead.